Tutorials/Tutorial27_PostProcessing/assets/shaders/ComputeMotionVectors.fx

```diff
@@ -0,0 +1,93 @@
+    ret[0][2] = (n22 * n34 * n41 - n24 * n32 * n41 + n24 * n31 * n42 - n21 * n34 * n42 - n22 * n31 * n44 + n21 * n32 * n44) * idet;
+    ret[0][3] = (n23 * n32 * n41 - n22 * n33 * n41 - n23 * n31 * n42 + n21 * n33 * n42 + n22 * n31 * n43 - n21 * n32 * n43) * idet;
+
+    ret[1][0] = t12 * idet;
+    ret[1][1] = (n13 * n34 * n41 - n14 * n33 * n41 + n14 * n31 * n43 - n11 * n34 * n43 - n13 * n31 * n44 + n11 * n33 * n44) * idet;
+    ret[1][2] = (n14 * n32 * n41 - n12 * n34 * n41 - n14 * n31 * n42 + n11 * n34 * n42 + n12 * n31 * n44 - n11 * n32 * n44) * idet;
+    ret[1][3] = (n12 * n33 * n41 - n13 * n32 * n41 + n13 * n31 * n42 - n11 * n33 * n42 - n12 * n31 * n43 + n11 * n32 * n43) * idet;
+
+    ret[2][0] = t13 * idet;
+    ret[2][1] = (n14 * n23 * n41 - n13 * n24 * n41 - n14 * n21 * n43 + n11 * n24 * n43 + n13 * n21 * n44 - n11 * n23 * n44) * idet;
+    ret[2][2] = (n12 * n24 * n41 - n14 * n22 * n41 + n14 * n21 * n42 - n11 * n24 * n42 - n12 * n21 * n44 + n11 * n22 * n44) * idet;
+    ret[2][3] = (n13 * n22 * n41 - n12 * n23 * n41 - n13 * n21 * n42 + n11 * n23 * n42 + n12 * n21 * n43 - n11 * n22 * n43) * idet;
+
+    ret[3][0] = t14 * idet;
+    ret[3][1] = (n13 * n24 * n31 - n14 * n23 * n31 + n14 * n21 * n33 - n11 * n24 * n33 - n13 * n21 * n34 + n11 * n23 * n34) * idet;
+    ret[3][2] = (n14 * n22 * n31 - n12 * n24 * n31 - n14 * n21 * n32 + n11 * n24 * n32 + n12 * n21 * n34 - n11 * n22 * n34) * idet;
+    ret[3][3] = (n12 * n23 * n31 - n13 * n22 * n31 + n13 * n21 * n32 - n11 * n23 * n32 - n12 * n21 * n33 + n11 * n22 * n33) * idet;
+
+    return ret;
+}
+
+float2 ComputeMotionVectorsPS(FullScreenTriangleVSOutput VSOut) : SV_Target0
+{
+    float Depth = g_TextureDepth.Load(uint3(VSOut.f4PixelPos.xy, 0));
+    uint ObjectIdx = g_TextureObjectID.Load(uint3(VSOut.f4PixelPos.xy, 0));
+
+    float4x4 CurrWorldMatrix = g_CurrObjectTransforms[ObjectIdx];
+    float4x4 PrevWorldMatrix = g_PrevObjectTransforms[ObjectIdx];
+
+    float4 NDC1 = float4(VSOut.f2NormalizedXY, DepthToNormalizedDeviceZ(Depth), 1.0f);
+
+    float4 Position = mul(NDC1, g_CurrCamera.mViewProjInv);
+    Position /= Position.w;
+
+    // For SDF (Signed Distance Field) transformations, we store matrices that transitioning from World Space to Local Space.
+    // Therefore, we do not invert the `CurrWorldMatrix`.
+    // However, we do invert the `PrevWorldMatrix` as it's necessary to transform the 'Position' point back into World Space.
+
+    // In the case of polygonal geometry,  you'll likely store matrices that transition from Local Space to World Space.
+    // Therefore, for this shader to work in such scenarios, you need to swap the corresponding matrix inversion.
+    // In other words, invert the `CurrWorldMatrix` and do not invert the `PrevWorldMatrix`
+
+    Position = mul(Position, CurrWorldMatrix);
+    Position = mul(Position, InverseMatrix(PrevWorldMatrix));
```

Comment:
Why do you inverse matrix in the shader??
`g_PrevObjectTransforms` can be inversed on the host.